pkg/controller/plan/adapter/ova/validator.go

```diff
@@ -111,10 +111,8 @@ func (r *Validator) StorageMapped(vmRef ref.Ref) (ok bool, err error) {
 		return
 	}
 
-	for _, disk := range vm.Disks {
-		if !r.plan.Referenced.Map.Storage.Status.Refs.Find(ref.Ref{ID: disk.ID}) {
-			return
-		}
+	if len(vm.Disks) == 0 {
+		return
```

Comment:
why do we need this part?

Reply:
validate that the vm has disks at all, in case its empty do we want to allow the mapping?

Reply:
if a vm has no disks then it doesn't need mappings - every mapping would be valid
we are generally not interested in diskless vms but if we want to block it, it needs to be a different validation, not of the mapping

Reply:
In that case we don't really need to check anything here, will always be true.